### PR TITLE
MOTOR-698 Motor 2 to 3 migration guide

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

File: doc/migrate-to-motor-3.rst [hunk 1]
```diff
@@ -0,0 +1,456 @@
...
+
+Removed :meth:`~motor.motor_tornado.MotorClient.unlock`. Run the
+`fsyncUnlock command`_ directly with
+:meth:`~motor.motor_tornado.MotorDatabase.command` instead. For example::
+
+     await client.admin.command('fsyncUnlock')
+
+.. _fsyncUnlock command: https://mongodb.com/docs/manual/reference/command/fsyncUnlock/
+
+MotorClient.is_locked is removed
```
Comment (ShaneHarvey) [inline doc/migrate-to-motor-3.rst:122]: Motor 2 doesn't support is_locked so we can remove this one.

File: doc/migrate-to-motor-3.rst [hunk 2]
```diff
@@ -0,0 +1,456 @@
...
+  if database:
+
+Can be changed to this::
+
+  if database is not None:
+
+You must now explicitly compare with None.
+
+
+MotorCollection
```
Comment (ShaneHarvey) [inline doc/migrate-to-motor-3.rst:290]: Missing a mention of reindex.

File: doc/migrate-to-motor-3.rst [hunk 3]
```diff
@@ -0,0 +1,456 @@
...
+
+  cursor = await coll.find({}, min={'x', min_value})
+
+can be changed to this::
+
+  cursor = await coll.find({}, min={'x', min_value}, hint=[('x', ASCENDING)])
+
+MotorCollection.__bool__ raises NotImplementedError
+...................................................
+:class:`~`~motor.motor_tornado.MotorCollection` now raises an error upon evaluating
```
Comment (ShaneHarvey) [inline doc/migrate-to-motor-3.rst:355]:
```
:class:`~`~motor...
```
->
```
:class:`~motor...
```

File: doc/changelog.rst [hunk 4]
```diff
@@ -6,12 +6,90 @@ Changelog
 Motor 3.0
 ---------
 
-Motor 3.0 requires PyMongo 4.0+ and Python 3.7+.
-
+Motor 3.0 adds support for PyMongo 4.0+.  It inherits a number
+of breaking API changes from PyMongo 4.0.
```
Comment (ShaneHarvey) [inline doc/changelog.rst:10]: Let's link to the Motor migration guide here.

File: doc/changelog.rst [hunk 5]
```diff
@@ -6,12 +6,90 @@ Changelog
 Motor 3.0
```
Comment (ShaneHarvey) [inline doc/changelog.rst:6]: Are there any notable _new_ features to document?

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: doc/migrate-to-motor-3.rst [hunk 6]
```diff
@@ -0,0 +1,456 @@
...
+deprecated functions and methods are being used in your application::
+
+  python -Wd <your application>
+
+Warnings can also be changed to errors::
+
+  python -Wd -Werror <your application>
+
+Note that there are some deprecation warnings raised by Motor itself for
+APIs that are deprecated but not yet removed, like :meth:`~motor.motor_tornado.MotorCusor.fetch_next`.
```
[the remark this answers is not in the document]
Comment (blink1073) [reply doc/migrate-to-motor-3.rst:47]: Fixed

File: doc/migrate-to-motor-3.rst (hunk 1 above)
Comment (blink1073) [reply doc/migrate-to-motor-3.rst:122]: Done

File: doc/migrate-to-motor-3.rst (hunk 2 above)
Comment (blink1073) [reply doc/migrate-to-motor-3.rst:290]: Added

File: doc/migrate-to-motor-3.rst (hunk 3 above)
Comment (blink1073) [reply doc/migrate-to-motor-3.rst:355]: Fixed

File: doc/changelog.rst (hunk 4 above)
Comment (blink1073) [reply doc/changelog.rst:10]: Done

File: doc/changelog.rst (hunk 5 above)
Comment (blink1073) [reply doc/changelog.rst:6]: Added applicable "notable improvements" from PyMongo 4.0 and 4.1

File: doc/changelog.rst [hunk 7]
```diff
@@ -7,90 +7,138 @@ Motor 3.0
...
+- Added the ``maxConnecting`` URI and
+  :class:`~motor.motor_tornado.MotorClient` keyword argument.
+- :class:`~motor.motor_tornado.MotorClient` now accepts a URI and keyword
+  argument `srvMaxHosts` that limits the number of mongos-like hosts a client
+  will connect to. More specifically, when a mongodb+srv:// connection string
+  resolves to more than `srvMaxHosts` number of hosts, the client will randomly
+  choose a `srvMaxHosts` sized subset of hosts.
+- Added :attr:`motor.motor_tornado.MotorClient.options` for read-only access
+  to a client's configuration options.
+- Added support for the ``comment`` parameter to all helpers. For example see
```
Comment (ShaneHarvey) [inline doc/changelog.rst:109]: commit and let support require PyMongo 4.1 so should we bump our min version to 4.1?

File: doc/migrate-to-motor-3.rst [hunk 8]
```diff
@@ -0,0 +1,461 @@
...
+
+The following client configuration option helpers are removed:
+- :attr:`~motor.motor_tornado.MotorClient.event_listeners`.
+- :attr:`~motor.motor_tornado.MotorClient.max_pool_size`.
+- :attr:`~motor.motor_tornado.MotorClient.min_pool_size`.
+- :attr:`~motor.motor_tornado.MotorClient.max_idle_time_ms`.
+- :attr:`~motor.motor_tornado.MotorClient.local_threshold_ms`.
+- :attr:`~motor.motor_tornado.MotorClient.server_selection_timeout`.
+- :attr:`~motor.motor_tornado.MotorClient.retry_writes`.
+- :attr:`~motor.motor_tornado.MotorClient.retry_reads`.
```
Comment (ShaneHarvey) [inline doc/migrate-to-motor-3.rst:157]: This list doesn't render: https://motor--166.org.readthedocs.build/en/166/migrate-to-motor-3.html#motorclient-event-listeners-and-other-configuration-option-helpers-are-removed

File: doc/changelog.rst [hunk 9]
```diff
@@ -6,12 +6,138 @@ Changelog
...
+- Comparing two :class:`~motor.motor_tornado.MotorClient` instances now
+  uses a set of immutable properties rather than
+  :attr:`~motor.motor_tornado.MotorClient.address` which can change.
+- Removed the `disable_md5` parameter for :class:`~gridfs.GridFSBucket` and
+  :class:`~gridfs.GridFS`. See :ref:`removed-gridfs-checksum` for details.
+- PyMongoCrypt 1.2.0 or later is now required for client side field level
+  encryption support.
+
+Notable improvements
+....................
```
Comment (ShaneHarvey) [inline doc/changelog.rst:96]: This must use `~`.

File: doc/changelog.rst [hunk 10]
```diff
@@ -6,12 +6,138 @@ Changelog
...
+  context.
+- :meth:`~motor.motor_tornado.MotorCollection.aggregate` now supports
+  $merge and $out executing on secondaries on MongoDB >=5.0.
+  aggregate() now always obeys the collection's :attr:`read_preference` on
+  MongoDB >= 5.0.
+- :meth:`gridfs.grid_file.GridOut.seek` now returns the new position in the file, to
+  conform to the behavior of :meth:`io.IOBase.seek`.
+
+Issues Resolved
+...............
```
Comment (ShaneHarvey) [inline doc/changelog.rst:135]: This must use `~`.

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: doc/changelog.rst (hunk 7 above)
Comment (blink1073) [reply doc/changelog.rst:109]: Sure, as a separate ticket?

File: doc/changelog.rst (hunk 10 above)
Comment (blink1073) [reply doc/changelog.rst:135]: Fixed

File: doc/changelog.rst (hunk 7 above)
Comment (ShaneHarvey) [reply doc/changelog.rst:109]: Yeah separately makes sense.

File: doc/migrate-to-motor-3.rst (hunk 8 above)
Comment (blink1073) [reply doc/migrate-to-motor-3.rst:157]: fixed

File: doc/changelog.rst (hunk 9 above)
Comment (blink1073) [reply doc/changelog.rst]: Done

File: doc/changelog.rst (hunk 10 above)
Comment (blink1073) [reply doc/changelog.rst]: Done

**Review by ShaneHarvey — APPROVED** [review]

**Review by juliusgeo — APPROVED** [review]
LGTM!

**Pull request merged**